configurable_lti_consumer/configurable_lti_consumer.py

```
@@ -28,32 +35,38 @@ def set_editable_field_names(self):
         Remove from editable_field_names list fields that we already have configured
         """
         self.editable_field_names = list(self.editable_field_names)
-        name = self.description
-        for field, _ in self.get_configuration(name)['default_values'].items():
-            self.editable_field_names.pop(self.editable_field_names.index(field))
+        configuration = self.get_configuration(self.lti_id)
+        if configuration:
+            for field, _ in configuration['default_values'].items():
+                self.editable_field_names.pop(self.editable_field_names.index(field))
 
     @classmethod
     def get_configuration(cls, name):
         """
         Retrieving component subclass configuration from Django settings
         """
-        for component in settings.CONFIGURABLE_XBLOCKS_SETTINGS["components"]:
-            if component["module"] == "lti_consumer":
-                conf = [
-                    subclass for subclass in component["subclasses"]
-                    if subclass["name"] == name]
-                if conf:
-                    return conf[0]
-        return None
+        if getattr(settings, "CONFIGURABLE_XBLOCKS_SETTINGS"):
+            for component in settings.CONFIGURABLE_XBLOCKS_SETTINGS["components"]:
+                if component["module"] == "lti_consumer":
+                    conf = [
+                        subclass for subclass in component["subclasses"]
+                        if subclass["name"] == name]
+                    if conf:
+                        return conf[0]
+                    else:
+                        logger.warning(
+                            "Configuration '{name}' does not exist".format(name=name))
+        else:
+            logger.warning("CONFIGURABLE_XBLOCKS_SETTINGS settings are not set")
```

let's break and raise a ConfigurationError instead of just logging a warning because it is the admin sys reponsibility to fix it before starting the server. WDYT?